http-api/src/v1/projects.rs

```diff
@@ -720,3 +720,259 @@ fn resolve_revisions(
         revisions,
     }
 }
+
+mod setup {
+    use std::env;
+    use std::path::Path;
+
+    use git2::Oid;
+
+    use librad::crypto::keystore::crypto::{Pwhash, KDF_PARAMS_TEST};
+    use librad::crypto::keystore::pinentry::SecUtf8;
+    use librad::crypto::BoxedSigner;
+    use librad::git::identities::Project;
+    use librad::git::util;
+    use librad::git_ext::tree;
+    use librad::profile::{Profile, LNK_HOME};
+
+    use radicle_common::{keys, person, profile, project, test};
+
+    #[allow(dead_code)]
+    pub fn profile() -> (Profile, BoxedSigner, Project, Oid) {
```

Comment:
- wouldn't it make more sense if we call this `environment`, usage would then be `setup::environment()` and `teardown::environment()`.

- am I right to skip using `teardown` here due to concurrency?

- `assay` is not applicable here, is it?

- we can call setup once with `std::sync::Once`, is that something we'd be interested in?

Reply:
Yeah renaming makes sense. Teardown is not needed since it cleans up everything at the end of the function afaict.

Reply:
it doesn't, look:
```
⮀ ls /tmp/rad/home/api/
012eb9e9-323f-47f1-8723-5cc0c25e1c46/  229dbdff-9cb3-419b-89f6-efd198931afc/  3b3eeee3-4048-4dfa-bf86-45d99d511311/  4f8540d0-76c8-4a8b-a18a-21a6a70d25c3/  6bbfaef6-6cab-407e-a8f8-db5a536a3896/  8875415f-2cd4-4483-b3bd-a6ddd1377f60/  b4155c08-3132-484c-b3d8-22071402114a/  c58a9c69-7225-49d0-8601-ecaf32989d53/  e591f9e2-4575-4f73-8dd7-1f779ddb5b80/
0181ed78-cd11-4a9c-ab1f-166a020e9a50/  22ec3547-e0dd-4792-b412-4ba631374524/  3bad31eb-46f8-4836-8138-5b4584164fde/  50ea4dab-4a3e-460c-b6d3-fb6c409cd180/  6ee46bc1-ac70-42b4-a385-1720e93dd2b2/  88e89f77-1b1e-4bff-a04a-761d609bf39d/  b7672099-fd02-4796-bd12-dab29f4c0444/  c83bf037-de9d-46fc-be8c-da10b0468721/  e76d59bb-7f43-4b6f-ae5b-27c325da311b/
06a07c36-180d-47ee-a8f0-ff8c9f159cdc/  2d46d08b-1f18-4a95-9f24-92e18026c1be/  3ddd1dd1-39de-4ccd-a2be-60f552f02ef2/  593e47f4-ae70-4f5e-97a3-13639f9e4dcc/  731caa4b-cf26-4c12-9319-082ad07bef77/  89490997-b505-45a5-8ab3-ce72a4d7ad2a/  b8159b5c-2c4f-4d59-8b48-a28342f56e25/  cbb81cb7-f1b6-45d8-ae79-d159038663da/  e8409be1-5cd1-4780-8da5-e06109475eb8/
0852a8d2-bd75-4b11-8d49-dab680889549/  2e8bbe2c-3d8b-46d1-afbc-671f6d0d7ea1/  3ec386ac-1899-4585-98b3-010548e699bf/  5a6f9caf-eb1d-4925-8e03-a261f370a4c0/  74de4eaa-fe79-4d95-a8ab-2c0062f34217/  8aaf72d7-a0a2-451b-914c-cbec67eb0f9c/  ba4a34d2-62c5-4fbc-b6cc-a38c71eeed85/  d1f12247-1c82-49a6-9c60-73a526c431a9/  ebdb9a8f-e14c-4f2b-b816-be78700720e1/
0caa4ecd-c14f-40ac-940b-de71a28aa0b4/  347d8215-5939-4f01-9a4e-2cd0cfbb4455/  3f696799-22e0-4aef-ae9f-bbbbd90d9ccc/  5af7bdc4-6efe-4215-97fe-631025521809/  7a4c6e69-4cdc-4e68-a127-b43641ef3b01/  8e815715-f976-4cfe-8688-c2377064f505/  bb63942f-02b4-4d35-8edd-7bca835e5798/  d3602b0e-293e-4f58-9eac-61be12f81d75/  ec51e0ce-7ec2-4977-bbd3-e9f52ff04a57/
0eed204c-6f83-463e-84cf-985873bfd796/  35ec2164-9a37-4b25-831f-93860c47c2bc/  3fb5829e-8d9c-47c5-9a67-3ee2233d8c03/  5ec9e82b-739e-4dd7-96f6-0d288b28c7fc/  7e63ea22-d74a-4f51-a934-1b0d95de52ae/  9729dadd-e7ee-4f7f-bf96-3a69067669f8/  bb84983f-1ba4-4d81-9321-6204faf4ab78/  d4c7f5f9-f6fb-441c-86dc-07793054f5d1/  f25aa64e-4a22-4ef7-994c-9f4048919ee3/
128a0bc4-2fcc-4ed3-80fa-a4a0abe9d170/  37311270-e9c3-490a-8b38-fe15e24dcf1b/  40047ae9-ccda-4b34-9a9a-06b97c8e23d1/  5fa24f57-e6d9-42aa-8fcf-03e0fd893638/  7f13f527-b915-421c-9355-de1723cd8ef9/  98b3b801-fdd0-4996-a5ad-0c75eb465bbf/  bb8763b0-d1c9-46ae-a588-80886512acb4/  d55fa684-1f50-409c-ad56-ddd9be38ec60/  f3621a8c-ce31-48bd-a49a-12021d0de825/
13feae74-594b-414e-a7e8-d106cfc4a6f5/  386a690c-8143-4e98-8b1c-239b2ab32139/  422713ac-a369-428c-83d4-82d68d7d4450/  6187eb39-1f29-4aad-9ac6-128c60abff9f/  80ba144e-9e95-43fa-b7c1-685edd44fcad/  9f09d16e-f4c3-46b9-889d-95a7009aad9f/  bd235ac7-323c-4f48-bfd6-99d935e7da97/  d702453a-6531-450f-9427-b7ce1a7e43c5/  f415d527-a274-4a32-b900-c52b8dedd706/
1a9dba3c-7dec-4932-897c-b78b364f8434/  389137f2-d720-41e1-93e7-d40127f07f0a/  46157029-5939-4bad-a81f-f2da528a74e5/  62737d1a-e87d-4080-ad42-552227347d25/  842bd79b-c7c0-4101-85d0-6818ab5ab6dd/  a090dd1e-3258-4bb8-baf8-7741c6e577db/  be739d7c-936e-4220-9f05-f6667424684a/  d9a20c75-ba0d-4ff9-a5f2-c9fc3b7db480/  f719799b-e832-4abc-b9dd-c2b477fc86fd/
1ab387b4-2e1d-4ba5-9576-147b2b4c9200/  390f38be-c9b5-405b-9be8-03090860fbe8/  4b5a34ea-e532-4381-b476-9719157e476a/  692f49a8-d7fc-4c85-9af7-4cf8aaa8f81a/  86573dfc-5e47-4fa2-ab18-51aed705ba03/  a666a502-3d82-4745-a628-fc628fc1f874/  bfedf104-cc2a-4f40-be54-5233b0a8fed2/  dc1262de-f6bf-44d6-af45-87c74564aba8/  f7cf6d11-47c3-4bae-b9bb-1c3dbaafa9fa/
1bae07ec-f458-441b-8e96-dab0553f6b6c/  3a3451c5-2f60-4ce4-8797-973cf337522c/  4dc3bd70-97cd-4d5e-a5f1-dc274abcd92a/  6b08600e-766c-4fb6-9905-2c6de72cb77a/  8805b732-86fd-4b46-8372-921a01cc4dd7/  active_profile                         c1648562-8923-4fe1-9aeb-6d5643c2e109/  df10a843-b09f-4de3-b468-420d9ab1bd26/  fcf7de3c-a213-416f-b18c-6b7da2440ed2/
```

Reply:
can use [`std::fs::remove_dir_all`](https://doc.rust-lang.org/std/fs/fn.remove_dir_all.html) to manually remove these though.